providers/class-two-factor-backup-codes.php

```diff
@@ -56,7 +56,7 @@ public function register_rest_routes() {
 				'methods'             => WP_REST_Server::CREATABLE,
 				'callback'            => array( $this, 'rest_generate_codes' ),
 				'permission_callback' => function( $request ) {
-					return current_user_can( 'edit_user', $request['user_id'] );
+					return Two_Factor_Core::rest_api_can_edit_user_and_update_two_factor_options( $request['user_id'] );
```

Comment:
That's a very good point, and something I was considering here.

For example, should 2FA confirmation be needed to change your own password? Seems like that would be a reasonable suggestion! but right now with how this is implemented that's not really possible.

But on the other hand, we can't really override the capability checks, because they include very little context of "what" the capability is being checked for. Is it being checked to see if the user edit screen is being displayed? or is it being checked to see if the user can be edited?

I was thinking about this, and one option would be to capture all `edit_user` capability checks on the REST API between `rest_request_before_callbacks` and `rest_dispatch_request`, but that's *only* rest-api checks, it doesn't cover any non-api-based methods - which is why I ended up with this callback.